src/cdomains/valueDomain.ml

```diff
@@ -27,36 +29,48 @@ sig
   val smart_join: (exp -> int64 option) -> (exp -> int64 option) -> t -> t ->  t
   val smart_widen: (exp -> int64 option) -> (exp -> int64 option) ->  t -> t -> t
   val smart_leq: (exp -> int64 option) -> (exp -> int64 option) -> t -> t -> bool
+  val is_immediate_type: typ -> bool
+  val bot_value: typ -> t
+  val init_value: typ -> t
+  val top_value: typ -> t
+  val zero_init_value: typ -> t
 end
 
 module type Blob =
 sig
   type value
   type size
-  include Lattice.S with type t = value * size
+  type origin
+  include Lattice.S with type t = value * size * origin 
 
   val make: value -> size -> t
   val value: t -> value
   val size: t -> size
   val invalidate_value: Q.ask -> typ -> t -> t
 end
 
-module Blob (Value: S) (Size: IntDomain.S) =
+module AllocOrigin = IntDomain.MakeBooleans (
+  struct
+    let truename = "Malloc"
+    let falsename = "Calloc"
+  end)
```

Comment:
Is this intended to be the boolean lattice, where `Calloc ⊑ Malloc`, and not a flat lattice?

Reply:
You are right, they should be incomparable. Now it's a fake lattice of booleans.